addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -159,6 +162,12 @@ var ImageDialog = Widget.extend({
                 return;
             }
             self.page += $target.hasClass('previous') ? -1 : 1;
+            if ((self.page + 1) * self.per_page === self.records.length) {
```

Comment:
Not sure to understand the condition. If you already have the record of the next page then you fetch them again?

Reply:
If the last record on the current page is the last record overall, then I fetch the next page. `this.page` is 0 indexed, hence the `+ 1`. I think I'll just add a field to keep track of when I've read all matching records, it'll be much easier to understand that way.